app_test.go

```diff
@@ -1803,3 +1803,31 @@ func TestApp_GetRoutes(t *testing.T) {
 		utils.AssertEqual(t, name, route.Name)
 	}
 }
+
+func TestRouteName(t *testing.T) {
+	home, named := "home", "named"
+	app := New()
+
+	app.Use(func(c *Ctx) error {
```

Comment:
what happens if we define 2 middlewares with names one after the other?

can you add this to your test case 

and maybe one middleware which has no name

Reply:
Thanks for quick review. Yes I have 3 test cases, two with names & one unnamed - they all pass. Without looping over the stack the names were overwriting each other so this was the only way I found to resolve this problem

Reply:
the case for the middelwares with prefix matching is a little bit different

can you add additional route declarations
```go
app.Use(func(c *Ctx) error {
    return c.Next()
}).Name("corsMW")

app.Use(func(c *Ctx) error {
    return c.Next()
}).Name("compressMW")

app.Use(func(c *Ctx) error {
    return c.Next()
})// no name
```

and check the expectation

Reply:
can you include these cases in our tests

would like to prevent that the different declarations of the different routes with the same path influence each other.

Reply:
Sure will add in for review.

But to be honest, I think you'll need to reject this PR. Looping over the stack is going to cause the paths to overwrite each other so I do think this will need a different solution.

Reply:
Right I've added in as requested, but I'm less confident this meets expectations now so understand if you prefer to reject.